ndcube/ndcube.py

```diff
@@ -444,42 +444,56 @@ def quantity(self):
         return u.Quantity(self.data, self.unit, copy=_NUMPY_COPY_IF_NEEDED)
 
     def _generate_world_coords(self, pixel_corners, wcs, needed_axes=None, *, units):
-        # Create meshgrid of all pixel coordinates.
-        # If user wants pixel_corners, set pixel values to pixel corners.
+        # This will generate only the coordinates that are needed if there is no correlation within the WCS
+        # This bypasses the entire rest of the function below which works out the full set of coordinates
+        # This only works for WCS that have the same number of world and pixel dimensions
```

Comment:
This probably shouldn't be a limitation but right now I can't work out a good idea.

Reply:
Can you elaborate on what the difficulty is here?

Reply:
Too stupid to work out the line of code to overcome this limiation.

Reply:
OK.  Put another way, why does enforcing this limitation make the problem simpler?  If you just remove this condition, so long as `np.sum(wcs.axis_correlation_matrix[needed_axes]) == 1`, does it still work as intended?

Reply:
Would [this util function](https://github.com/sunpy/ndcube/blob/main/ndcube/utils/wcs.py#L111) solve that problem?

Reply:
Maybe it does but I don't see how.

Reply:
It might but I can't work out how.